### PR TITLE
[PW-2461] OneClick payment methods

### PR DESCRIPTION
**Pull request by acampos1916**
## Summary
After saving the payment details for future use this PR enables the payment with saved methods.

## Tested scenarios
Payments with stored card details, 3DS1/3DS2, optional and required CVC

**Fixed issue**:  PW-2461

### CODE REVIEW

File: src/Handlers/OneClickPaymentMethodHandler.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace Adyen\Shopware\Handlers;
+
+use Shopware\Core\Checkout\Payment\Cart\PaymentHandler\AsynchronousPaymentHandlerInterface;
+
+class OneClickPaymentMethodHandler extends AbstractPaymentMethodHandler implements AsynchronousPaymentHandlerInterface
+{
+
+    public static function getPaymentMethodCode()
+    {
+        return 'storedPaymentMethods';
```
Comment (cyattilakiss) [inline src/Handlers/OneClickPaymentMethodHandler.php:35]: can this conflict with recurring stored payment methods in the future if we add recurring as well?

File: src/Resources/app/storefront/src/checkout/checkout.plugin.js [hunk 2]
```diff
@@ -166,6 +171,54 @@ export default class CheckoutPlugin extends Plugin {
         }
     }
 
+    renderStoredPaymentMethod(paymentMethod) {
+        //  if the container doesn't exits don't try to render the component
+        const paymentMethodContainer = $('[data-adyen-payment-method="' + this.paymentMethodTypeHandlers["oneclick"] + '"]');
+
+        // container doesn't exist, something went wrong on the template side
+        // If payment method doesn't have details, just skip it
```
Comment (cyattilakiss) [inline src/Resources/app/storefront/src/checkout/checkout.plugin.js:179]: Not true anymore, comment can be removed
```suggestion
```

File: src/Resources/app/storefront/src/checkout/checkout.plugin.js [hunk 3]
```diff
@@ -166,6 +171,54 @@ export default class CheckoutPlugin extends Plugin {
...
+        });
+
+        /*Use the storedPaymentMethod object and the custom onChange function as the configuration object together*/
+        const configuration = Object.assign(paymentMethod, {
+            onChange: this.onPaymentMethodChange.bind(this)
+        });
+
+        if (paymentMethod.type === 'scheme') {
+            configuration.enableStoreDetails = true;
+        }
```
Comment (cyattilakiss) [inline src/Resources/app/storefront/src/checkout/checkout.plugin.js:202]: Is it needed for oneclick payment methods?
I think it's only required for payment methods where you do the oneoff payment and using that payment you create the token at Adyen. Therefore for already stored payment methods I believe this config is not needed.

```suggestion
```

File: src/Handlers/OneClickPaymentMethodHandler.php (hunk 1 above)
Comment (acampos1916) [reply src/Handlers/OneClickPaymentMethodHandler.php:35]: We can create a Recurring payment method handler that uses the same code if necessary, as long as only OneClick shows in the checkout. It wouldn't be a conflict.

Comment (cyattilakiss) [reply src/Handlers/OneClickPaymentMethodHandler.php:35]: sounds good :)

**Review by msilvagarcia — APPROVED** [review]

File: src/Subscriber/PaymentSubscriber.php [hunk 4]
```diff
@@ -215,12 +218,11 @@ public function onCheckoutConfirmLoaded(PageLoadedEvent $event)
...
                     ),
-                    'paymentMethodsResponse' => json_encode(
-                        $this->paymentMethodsService->getPaymentMethods($salesChannelContext)
-                    ),
+                    'paymentMethodsResponse' => json_encode($paymentMethodsResponse),
                     'orderId' => $orderId,
                     'stateDataPaymentMethod' => $stateDataPaymentMethod,
-                    'pluginId' => $adyenPluginId
+                    'pluginId' => $adyenPluginId,
+                    'storedPaymentMethods' => $paymentMethodsResponse['storedPaymentMethods'] ?? []
```
Comment (cyattilakiss) [inline src/Subscriber/PaymentSubscriber.php:225]: don't you need to json_encode this as well?

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**